fix: Add a test, fix the errors it revealed

necaris opened the pull request:
With apologies to @tsdh for being the one to run into this -- fix the ~inheritance of font faces~ handling of fontification, and add a test with ERT to verify things work as expected.

Should take care of #1 , ~although as @tsdh points out this may not actually be the correct way to do this.~

necaris committed:
feat: Add basic test

Add a simple ERT test with a toy `.jjdescription` file (taken from Steve
Klabnik's tutorial[1]), and update the README to describe how to run it.

1: https://steveklabnik.github.io/jujutsu-tutorial/hello-world/describing-commits.html

necaris committed:
fix: Fontify more correctly, per docs

tsdh commented:
This works better.
<img width="581" height="188" alt="image" src="https://github.com/user-attachments/assets/b75e1551-a503-4d87-8855-8f73d9107079" />
Note that the middle JJ: line is not fontified.

Still, I think the approach of setting the `face` property in your match functions directly is not complying with the documentation.  Rather the docs of `font-lock-keywords` say that entries should have the form `(match-function . face-name)` to fontify the complete match of `match-function` with `face-name` or `(match-function 2 face-name)` to fontify the group 2 of `match-function`'s match-data with `face-name` (and several other variants).  In any case, the `match-function` only does the search and doesn't set properties itself.  I can't tell which possible problematic side-effects arise from your approach.

By the way, I think that comments are usually not fontified using the font-lock machinery but with syntactic fontification.  Most probably setting `comment-start` to `"JJ:` is enough to fontify those complete lines as comment.  You might still want to have a font-lock rule for fontifying the "headers" inside comments, like `change ID: xyz`.  The font-lock match functions can assume that syntactic fontification is already done, i.e., they can skip over comments.  I think, when your matchers are plain regexps, that's automatic.  In matcher functions, you could check with `(nth 4 (syntax-ppss))` if `(point)` is inside a comment.

Fontification is hard. ;-)

necaris committed:
Add GHA to run tests

necaris commented:
@tsdh thank you for the feedback! I've refactored the fontification to do it "properly" per the documentation, and added a couple more test cases. Does this behave correctly for you?

_Edit: As the tests are passing and this works correctly for me, this is an unambiguous improvement on `main` and I'll merge; @tsdh can you please close #1 if it works for you?_

tsdh commented:
Good job with 29baf85. Now it looks (almost) good.

<img width="588" height="196" alt="image" src="https://github.com/user-attachments/assets/40f7713d-0162-418c-b527-3b6419581908" />

Do you see that the "C" of "Change ID" is highlighted differently?  I guess the compete header up to ":" should be highlighted like that?

But that's just a minor glitch.  I'll close #1.
